tests/vm_networking_test.go

```diff
@@ -97,48 +99,66 @@ var _ = Describe("Networking", func() {
-		outboundVM = tests.NewRandomVMWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskCirros), "#!/bin/bash\necho 'hello'\n")
-		_, err = virtClient.VM(tests.NamespaceTestDefault).Create(outboundVM)
-		Expect(err).ToNot(HaveOccurred())
+			// Create and start outbound VM
+			outboundVM := tests.NewRandomVMWithEphemeralDiskAndUserdata(tests.RegistryDiskFor(tests.RegistryDiskCirros), "#!/bin/bash\necho 'hello'\n")
+			_, err = virtClient.VM(tests.NamespaceTestDefault).Create(outboundVM)
+			Expect(err).ToNot(HaveOccurred())
 
-		for _, networkVm := range []*v1.VirtualMachine{inboundVM, outboundVM} {
-			waitUntilVmReady(networkVm, tests.LoggedInCirrosExpecter)
+			for _, networkVm := range []*v1.VirtualMachine{inboundVM, outboundVM} {
+				waitUntilVmReady(networkVm, tests.LoggedInCirrosExpecter)
+			}
+
+			inboundVM, err = virtClient.VM(tests.NamespaceTestDefault).Get(inboundVM.Name, v13.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			expecter, _, err := tests.NewConsoleExpecter(virtClient, inboundVM, 10*time.Second)
+			Expect(err).ToNot(HaveOccurred())
+			defer expecter.Close()
+			resp, err := expecter.ExpectBatch([]expect.Batcher{
+				&expect.BSnd{S: "\n"},
+				&expect.BExp{R: "\\$ "},
+				&expect.BSnd{S: "screen -d -m nc -klp 1500 -e echo -e \"Hello World!\"\n"},
+				&expect.BExp{R: "\\$ "},
+				&expect.BSnd{S: "echo $?\n"},
+				&expect.BExp{R: "0"},
+			}, 60*time.Second)
+			log.DefaultLogger().Infof("%v", resp)
+			Expect(err).ToNot(HaveOccurred())
+
+			outboundVM, err = virtClient.VM(tests.NamespaceTestDefault).Get(outboundVM.Name, v13.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			return inboundVM, outboundVM
 		}
 
-		inboundVM, err = virtClient.VM(tests.NamespaceTestDefault).Get(inboundVM.Name, v13.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		expecter, _, err := tests.NewConsoleExpecter(virtClient, inboundVM, 10*time.Second)
-		Expect(err).ToNot(HaveOccurred())
-		defer expecter.Close()
-		resp, err := expecter.ExpectBatch([]expect.Batcher{
-			&expect.BSnd{S: "\n"},
-			&expect.BExp{R: "\\$ "},
-			&expect.BSnd{S: "screen -d -m nc -klp 1500 -e echo -e \"Hello World!\"\n"},
-			&expect.BExp{R: "\\$ "},
-			&expect.BSnd{S: "echo $?\n"},
-			&expect.BExp{R: "0"},
-		}, 60*time.Second)
-		log.DefaultLogger().Infof("%v", resp)
-		Expect(err).ToNot(HaveOccurred())
+		inboundVM, outboundVM = prepareVMs()
+		// same but without pod network definition
+		inboundVM_no_podnet, outboundVM_no_podnet = prepareVMs()
 
-		outboundVM, err = virtClient.VM(tests.NamespaceTestDefault).Get(outboundVM.Name, v13.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
+		for _, networkVm := range []*v1.VirtualMachine{inboundVM_no_podnet, outboundVM_no_podnet} {
+			// sanity check the default spec uses explicit pod network
+			Expect(networkVm.Spec.Domain.Devices.Interfaces).ToNot(BeZero())
+			Expect(networkVm.Spec.Networks).ToNot(BeZero())
+			networkVm.Spec.Domain.Devices.Interfaces = []v1.Interface{}
```

Comment:
wouldn't the vm be created at this point?
I'm probably missing something, but I can't see how this will affect the test.

Reply:
Yeah you are right, I don't know what I was thinking.

Reply:
I think I solved the issue in the latest version. Now I configure default Networks / Interfaces before creating the resources. I don't like the approach we take with all these New*VM* functions, so instead of creating yet another one helper, I pass a boolean flag into prepareVMs and act on it if it's true by adding pod interface to the spec.